graph/src/data/schema.rs

```diff
@@ -606,6 +815,193 @@ impl Schema {
             })
     }
 
+    fn validate_fulltext_directives(&self) -> Vec<SchemaValidationError> {
+        self.subgraph_schema_object_type()
+            .map_or(vec![], |subgraph_schema_type| {
+                subgraph_schema_type
+                    .directives
+                    .iter()
+                    .filter(|directives| directives.name.eq("fulltext"))
+                    .fold(vec![], |mut errors, fulltext| {
+                        self.validate_fulltext_directive_name(fulltext)
+                            .into_iter()
+                            .for_each(|err| errors.push(err));
+                        self.validate_fulltext_directive_language(fulltext)
+                            .into_iter()
+                            .for_each(|err| errors.push(err));
+                        self.validate_fulltext_directive_algorithm(fulltext)
+                            .into_iter()
+                            .for_each(|err| errors.push(err));
+                        self.validate_fulltext_directive_includes(fulltext)
+                            .into_iter()
+                            .for_each(|err| errors.push(err));
+                        errors
+                    })
+            })
+    }
+
+    fn validate_fulltext_directive_name(&self, fulltext: &Directive) -> Vec<SchemaValidationError> {
+        let name = match fulltext.argument("name") {
+            Some(Value::String(name)) => name,
+            _ => return vec![SchemaValidationError::FulltextNameUndefined],
+        };
+
+        let local_types: Vec<&ObjectType> = self
+            .document
+            .get_object_type_definitions()
+            .into_iter()
+            .collect();
+
+        // Validate that the fulltext field doesn't collide with any entity fields
```

Comment:
There is something we should validate instead of entity field collisions though:

For every entity type `Foo`, we generate top-level fields `foo` and `foos`. We have to check for collisions with these. This check should reuse the singular and plural field naming logic from the API schema module.